easistrain/EDD/angleCalibEDD.py

```diff
@@ -35,28 +35,41 @@ def angleCalibrationEDD(
     """Main function."""
 
     with h5py.File(fileRead, "r") as h5Read:  ## Read the h5 file of raw data
-        patternHorizontalDetector = h5Read[
+        nb_channels = h5Read[
             sample
             + "_"
             + str(dataset)
             + "_"
             + str(scanNumber)
             + ".1/measurement/"
             + nameHorizontalDetector
-        ][
-            ()
-        ]  ## pattern of horizontal detector
-        patternVerticalDetector = h5Read[
-            sample
-            + "_"
-            + str(dataset)
-            + "_"
-            + str(scanNumber)
-            + ".1/measurement/"
-            + nameVerticalDetector
-        ][
-            ()
-        ]  ## pattern of vertical detector
+        ].shape[
+            1
+        ]  # number of channels
+        patternHorizontalDetector = np.reshape(
+            h5Read[
+                sample
+                + "_"
+                + str(dataset)
+                + "_"
+                + str(scanNumber)
+                + ".1/measurement/"
+                + nameHorizontalDetector
+            ][()],
+            (nb_channels),
+        )  ## pattern of horizontal detector
```

Comment:
Why about doing this instead:

```python
patternHorizontalDetector = h5Read[
                sample
                + "_"
                + str(dataset)
                + "_"
                + str(scanNumber)
                + ".1/measurement/"
                + nameHorizontalDetector
            ][0]
```

instead of the reshape ?

Reply:
This should work as well and is a better choice. Then we will get rid of reading the shape --> increase the number of lines in the code :+1: 
I will do the modifs